vm.go

```diff
@@ -351,6 +354,70 @@ func (vm *VirtualMachineSpec) ExecuteCommand(command string) error {
 	return nil
 }
 
+// escapeArgs makes a Windows-style escaped command line from a set of arguments
+func escapeArgs(args []string) string {
+	escapedArgs := make([]string, len(args))
+	for i, a := range args {
+		escapedArgs[i] = windows.EscapeArg(a)
+	}
+	return strings.Join(escapedArgs, " ")
+}
+
+// RunCommand executes a command on the Virtual Machine
+func (vm *VirtualMachineSpec) RunCommand(command []string, user string) (output string, errOut string, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Minute)
+	defer cancel()
+	system, err := hcs.OpenComputeSystem(ctx, vm.ID)
+	if err != nil {
+		return "", "", err
+	}
+	defer system.Close()
+
+	params := &hcsschema.ProcessParameters{
+		CommandArgs:      command,
+		WorkingDirectory: "/",
+		User:             user,
+		Environment:      map[string]string{"PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"},
+		CreateStdInPipe:  false,
+		CreateStdOutPipe: true,
+		CreateStdErrPipe: true,
+		ConsoleSize:      []int32{0, 0},
+	}
+
+	if system.OS() == "windows" {
+		params.CommandLine = escapeArgs(command)
```

Comment:
>CommandLine [](http://example.com/codeflow?start=9&length=11)

if os is windows, do we need CommandArgs in ProcessParameters?  Can we have dedicate statements to initialize ProcessParameters per os type?

Reply:
we dont need CommandArgs for windows. will do..